plugins/coverbrowser.koplugin/main.lua

```diff
@@ -161,58 +162,132 @@ function CoverBrowser:addToMainMenu(menu_items)
     -- add Mosaic / Detailed list mode settings to File browser Settings submenu
     -- next to Classic mode settings
     if menu_items.filebrowser_settings == nil then return end
-    table.insert (menu_items.filebrowser_settings.sub_item_table, 4, {
+    table.insert (menu_items.filebrowser_settings.sub_item_table, 5, {
         text = _("Mosaic and detailed list settings"),
         separator = true,
         sub_item_table = {
             {
-                text = _("Items per page"),
-                help_text = _([[This sets the number of files and folders per page in display modes other than classic.]]),
+                text_func = function()
+                    local nb_cols_portrait = tonumber(BookInfoManager:getSetting("nb_cols_portrait") or 3)
+                    local nb_rows_portrait = tonumber(BookInfoManager:getSetting("nb_rows_portrait") or 3)
+                    return T(_("Items per page in mosaic portrait mode: %1 x %2"), nb_cols_portrait, nb_rows_portrait)
+                end,
                 -- Best to not "keep_menu_open = true", to see how this apply on the full view
-                callback = function()
+                callback = function(touchmenu_instance)
+                    local nb_cols_portrait = tonumber(BookInfoManager:getSetting("nb_cols_portrait") or 3)
+                    local nb_rows_portrait = tonumber(BookInfoManager:getSetting("nb_rows_portrait") or 3)
+                    local DoubleSpinWidget = require("/ui/widget/doublespinwidget")
+                    local widget = DoubleSpinWidget:new{
+                        title_text = _("Mosaic portrait mode"),
+                        width_factor = 0.6,
+                        left_text = _("Columns"),
+                        left_value = nb_cols_portrait,
+                        left_min = 2,
+                        left_max = 5,
```

Comment:
I [argued against going up to 30](https://github.com/koreader/koreader/issues/11186#issuecomment-1858623949) :) but may be 5 is too little ? I guess one could be happy with 8 or 10 on large devices ? There's still some limit with the font size and reability of text covers - I don't know what limit it would bring.

Reply:
Waiting for @mergen3107 with big screen limits.

Reply:
I guess it's mostly a matter of taste in the end, but I probably wouldn't go higher than 6 or 8?

Reply:
On larger screens (monitors/TVs) it could make sense, but on the flip side when more than that makes sense you might also want 4 pages at once or something.